src/lockfile/parse.js

```diff
@@ -314,9 +314,101 @@ export class Parser {
   }
 }
 
-export default function(str: string, fileLoc: string = 'lockfile'): Object {
-  str = stripBOM(str);
+const MERGE_CONFLICT_START = '<<<<<<<';
+const MERGE_CONFLICT_SEP = '=======';
+const MERGE_CONFLICT_END = '>>>>>>>';
+
+/**
+ * Extract the two versions of the lockfile from a merge conflict.
+ */
+
+export function extractConflictVariants(str: string): Array<string> {
+  const variants: Array<Array<string>> = [[], []];
+  const lines = str.split(/\n/g);
+
+  while (lines.length) {
+    const line = lines.shift();
+    if (line.startsWith(MERGE_CONFLICT_START)) {
+      // get the first variant
+      while (lines.length) {
+        const line = lines.shift();
+        if (line === MERGE_CONFLICT_SEP) {
```

Comment:
This feature looks awesome.

Curious if you all are wanting to support git's `diff3` merge conflict style (https://git-scm.com/docs/git-merge#_how_conflicts_are_presented). I think this code will break for users using `diff3` because the first variant would include both the real variant and the common parent ancestor of both conflicts. I wouldn't mind making a PR in the future for supporting `diff3`.

Sorry if I'm overlooking something.